src/commands/lint/lint.s3.paths.ts

```diff
@@ -0,0 +1,87 @@
+import { command, positional, string } from 'cmd-ts';
+
+import { CliInfo } from '../../cli.info.js';
+import { logger } from '../../log.js';
+import { config, registerCli, verbose } from '../common.js';
+import { imageryBucketNames, imageryCrs, imageryProducts, regions } from './lint.constants.js';
+
+export class MissingKeyError extends Error {
+  constructor(path: string) {
+    super(`Missing Key in Path: ${path}`);
+    this.name = 'MissingKeyError';
+  }
+}
+
+export class AdditionalKeyError extends Error {
+  constructor(path: string) {
+    super(`Additional arguments in path: ${path}`);
+    this.name = 'AdditionalKeyError';
+  }
+}
+
+export class InvalidKeyError extends Error {
+  constructor(type: string, value: string) {
+    super(`${type} not in ${type} list: ${value}`);
+    this.name = 'InvalidKeyError';
+  }
+}
+
+export const commandLintInputs = command({
+  name: 'lint-inputs',
+  description: 'Pretty print JSON files',
+  version: CliInfo.version,
+  args: {
+    config,
+    verbose,
+    path: positional({
+      type: string,
+      displayName: 'path',
+      description: 'JSON file to load inputs from, must be a JSON Array',
+    }),
+  },
+
+  async handler(args) {
+    registerCli(this, args);
+    const startTime = performance.now();
+
+    logger.info('LintInputs:Start');
+    if (!args.path) {
+      throw new Error('No path Provided');
+    }
+    logger.info({ target: args.path }, 'LintInputs:Info');
+    lintPath(args.path);
+    logger.info({ duration: performance.now() - startTime }, 'LintInputs:Done');
+  },
+});
+
+export function lintPath(path: string): void {
+  const [bucket, region, dataset, product, crs] = path.replace('s3://', '').split('/', 5);
+
+  if (bucket == null || region == null || dataset == null || product == null || crs == null) {
```

Comment:
[When there are not enough values to unpack (line 58), the values end up as `undefined` rather than `null`](https://www.typescriptlang.org/play?#code/FAYw9gdgzgLgBAbQGYEsBOsA0coFNwQAmAunALxwDkA7mGoZQHRQAOANijABSVUCUoSFDBtcjNmADmXPAUICgA), so we should probably `=== undefined` here.